root/static/scripts/common/components/EntityLink.js

```diff
@@ -155,47 +152,50 @@ const EntityLink = (props = {}) => {
   if (hover) {
     anchorProps.title = hover;
   }
-  content = <a {...anchorProps}>{isolateText(content)}</a>;
+  content = <a key={0} {...anchorProps}>{isolateText(content)}</a>;
 
   if (nameVariation) {
-    content = <span className="name-variation">{content}</span>;
+    content = <span className="name-variation" key={0}>{content}</span>;
   }
 
   if (!subPath && entity.editsPending) {
-    content = <span className="mp">{content}</span>;
+    content = <span className="mp" key={0}>{content}</span>;
   }
 
   if (!subPath && entityType === 'area') {
     let isoCodes = entity.iso_3166_1_codes;
     if (isoCodes && isoCodes.length) {
-      content = <span className={'flag flag-' + isoCodes[0]}>{content}</span>;
+      content = (
+        <span className={'flag flag-' + isoCodes[0]} key={0}>
+          {content}
+        </span>
+      );
     }
   }
 
   if (!showDisambiguation && !infoLink) {
     return content;
   }
 
-  return (
-    <Frag>
-      {content}
-      <If condition={showDisambiguation}>
-        <If condition={entityType === 'event'}>
-          <EventDisambiguation event={entity} />
-        </If>
-        <If condition={comment}>
-          <Comment className="comment" comment={comment} />
-        </If>
-        <If condition={entityType === 'area'}>
-          <AreaDisambiguation area={entity} />
-        </If>
-      </If>
-      <If condition={infoLink}>
-        {' '}
-        [<a href={infoLink}>{l('info')}</a>]
-      </If>
-    </Frag>
-  );
+  const parts = [content];
+
+  if (showDisambiguation) {
+    if (entityType === 'event') {
+      parts.push(<EventDisambiguation event={entity} key={1} />);
```

Comment:
Good idea, though I realized using an incremented number isn't a good idea to begin with. We should assign a unique key to each part, like "event_disambiguation" here, so that the keys are stable if some individual parts are added/removed.

Reply:
(That's why I hardcoded the numbers, but strings would make the intent more clear.)